src/batch/src/executor/join/nested_loop_join.rs

```diff
@@ -40,61 +37,29 @@ use crate::task::BatchTaskContext;
 ///
 ///
 /// High Level Idea:
-/// 1. Iterate tuple from probe table.
-/// 2. Concatenated with inner chunk, eval expression and get sel vector
-/// 3. Create new chunk with new sel vector and yield to upper.
+/// 1. Iterate tuple from left child.
+/// 2. Concatenated with right chunk, eval expression and get visibility bitmap
+/// 3. Create new chunk with visibility bitmap and yield to upper.
 pub struct NestedLoopJoinExecutor {
     /// Expression to eval join condition
     join_expr: BoxedExpression,
     /// Executor should handle different join type.
     join_type: JoinType,
+    /// Original output schema
+    original_schema: Schema,
+    /// Actual output schema
     schema: Schema,
     /// We may only need certain columns.
     /// output_indices are the indices of the columns that we needed.
     output_indices: Vec<usize>,
-    /// Return data chunk in batch.
-    chunk_builder: DataChunkBuilder,
-    /// Cache the chunk that has not been written into chunk builder
-    /// during join probing. Flush it in begin of execution.
-    last_chunk: Option<SlicedDataChunk>,
-    /// The data type of probe side. Cache to avoid copy too much.
-    probe_side_schema: Vec<DataType>,
-    /// Row-level iteration of probe side.
-    probe_side_source: RowLevelIter,
-    /// The table used for look up matched rows.
-    build_table: RowLevelIter,
-
-    /// Used in probe remaining iteration.
-    probe_remain_chunk_idx: usize,
-    probe_remain_row_idx: usize,
-
+    /// Left child executor (used in outer loop)
+    left_child: BoxedExecutor,
+    /// Right child executor (used in inner loop)
+    right_child: BoxedExecutor,
```

Comment:
We don't need a `build_table` to record matched rows? How to ensure join remain?

Reply:
> We don't need a `build_table` to record matched rows? How to ensure join remain?

The matched rows are yielded immediately in the nested loop. For right outer/semi/anti join, I use a bitmap to record matched rows of the right table, so remaining rows can be yielded when we go out of the loop.